### ISSUE TITLE
Symfony 4

### ISSUE THREAD

**Issue by lsv** [resquebundle/resque#31]
Is it actually been tested in a symfony 4 app?

These are the problems I have

* Commands not registered
* `$this->get('resque')` is not public, so can not be used in the controller or in commands, when I change it to dependency injection, templates can not be found.

Using dev-master branch

**Comment by PhilETaylor** [comment]
No testing in Symfony 4 has yet been done

Im currently using 3.4.2 in production only

I will be porting my production app to Symfony 4 early in the new year and will work on it then.

**Comment by lsv** [comment]
in  `services.xml`

Add these under `<services>`

```xml
<defaults autoconfigure="true" public="true" autowire="true" />
<prototype namespace="ResqueBundle\Resque\Command\" resource="../../Command">
    <tag name="console.command"/>
</prototype>
```

Now commands are working

**Comment by PhilETaylor** [comment]
I only have 5 mins in the office today, but I have branched to https://github.com/resquebundle/resque/tree/symfony4-compat so that Pull requests can be made against that branch

**Comment by danabrey** [comment]
Made a pull request that implements @lsv 's fix for command autowiring/autoconfiguring and Composer autoloading - #33

**Comment by kabudu** [comment]
Has there been any further movement regarding this issue?

**Comment by PhilETaylor** [comment]
And what further movement would you like? I dont have any active Symfony 4 projects in production due to other dependancies and so Im not working on symfony 4 at all. If you have contributions, then I'll be happy to test and merge them, and once there is a stable compatible symfony 4 version to tag a release in a higher SEMVER series.

**Comment by kabudu** [comment]
My comment was in response to this statement that you made 6 months ago:

```I will be porting my production app to Symfony 4 early in the new year and will work on it then.```

If you have no driver for working on it sooner, that's fair enough. I'm currently evaluating options and happened to come across your library, hence my curiosity!

**Comment by PhilETaylor** [comment]
Yeah I tried porting and realised I depend on much that has not bothered going to symfony 4 yet :( 

The multi award winning myJoomla.com service is powered by this library at the moment in production and it works very very well for us, we have some minor niggles with it but its stable and runs all our backend workers - normally 300 - across multiple servers

**Comment by kabudu** [comment]
The dependency rabbit hole can be vicious, I know. 😉 

I've used chrisboulton's PHP resque with Silex for a few years and it works well! However, I'm migrating away from Silex (given the impending deprecation) to Symfony 4 so really looking for options that are already Symfony 4 compatible without too much keyboard work!

**Comment by PhilETaylor** [comment]
Feel free to contribute, not many people have since I took over the maintenance of this package :) I would love to see it stable with symfony 4

I did spend some time looking at how to make Job's not "container aware" but to allow Dependancy injection the Symfony 3.4+ way, but I got lost in bootsrapping the container... would be also nice to get that finished one day but I just dont know enough about the bootstrap and prebuilt container to do it :(

**Comment by kabudu** [comment]
If I can find some spare time, I'll take a stab at making your library DI friendly with Symfony 4. 😉

**Comment by webignition** [comment]
Fixed in #29, available since [v2.0.6](https://github.com/resquebundle/resque/tree/2.0.6)
